fuzz/fuzz_targets/block.rs

```diff
@@ -132,10 +135,37 @@ fn memfd_create(name: &ffi::CStr, flags: u32) -> Result<RawFd, io::Error> {
 pub struct NoopVirtioInterrupt {}
 
 impl VirtioInterrupt for NoopVirtioInterrupt {
-    fn trigger(
-        &self,
-        _int_type: VirtioInterruptType,
-    ) -> std::result::Result<(), std::io::Error> {
+    fn trigger(&self, _int_type: VirtioInterruptType) -> std::result::Result<(), std::io::Error> {
         Ok(())
     }
 }
+
+fn wait_queue_event_processed(queue_evt: EventFd) {
```

Comment:
It's not clear to me how this new function is going to ensure the event is actually processed from the `virtio-block` thread. All it does is to check it received the event, this doesn't ensure about anything that could have happened on the `virtio-block` thread.

Reply:
The idea is using a "epoll" on the main thread to monitor the 'queue_evt' has been processed by the other thread (e.g. the virtio-block thread).  Before entering `epoll_wait()` on the main thread, both the "queue_evt" and "monitor_evt" have been kicked. The "epoll_wait()" on the main thread will report back either:
* 2 events: means both "queue_evt" and "monitor_evt" are not handled)
* 1 event : means one of the two events has been processed/read (which has to be the "queue_evt" because we only check the status of "monitor_evt" but never process/read it.

This is why we break on `num_events == 1` which is the sign the "queue_evt" has been processed. I should have put some comments around this function.

Reply:
Ok now I understand better what you tried to achieve. But I think you can simplify this design by providing a `timeout` value of `0` through the `epoll::wait()` call. This will make the `epoll_wait()` syscall non blocking, returning the amount of file descriptors that are ready. Therefore, you could simplify your code by breaking on `num_events == 0`, and no need to involve an extra `monitor_evt`.

Reply:
That's good input for using `epoll_wait()` with `timeout=0` to get rid of the `monitor_evt`.